x-pack/plugins/maps/public/components/single_field_select.tsx

```diff
@@ -7,35 +7,61 @@
 
 import _ from 'lodash';
 import React from 'react';
-
-import {
-  EuiComboBox,
-  EuiComboBoxProps,
-  EuiComboBoxOptionOption,
-  EuiHighlight,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiToolTip,
-} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { EuiComboBox, EuiComboBoxProps, EuiComboBoxOptionOption, EuiToolTip } from '@elastic/eui';
 import { FieldIcon } from '@kbn/react-field';
 import { DataViewField } from '@kbn/data-views-plugin/public';
+import { calculateWidthFromCharCount } from '@kbn/calculate-width-from-char-count';
+import { MIDDLE_TRUNCATION_PROPS } from '../../common/constants';
 
 function fieldsToOptions(
   fields?: DataViewField[],
-  isFieldDisabled?: (field: DataViewField) => boolean
+  isFieldDisabled?: (field: DataViewField) => boolean,
+  getFieldDisabledReason?: (field: DataViewField) => string | null
 ): Array<EuiComboBoxOptionOption<DataViewField>> {
   if (!fields) {
     return [];
   }
 
   return fields
     .map((field) => {
+      const FieldTypeIcon = field.type ? (
+        <FieldIcon type={field.type} fill="none" className="eui-alignMiddle" />
+      ) : null;
       const option: EuiComboBoxOptionOption<DataViewField> = {
         value: field,
         label: field.displayName ? field.displayName : field.name,
+        prepend: FieldTypeIcon,
       };
       if (isFieldDisabled && isFieldDisabled(field)) {
         option.disabled = true;
+
+        const disabledReason =
+          option.disabled && getFieldDisabledReason ? getFieldDisabledReason(option.value!) : null;
+
+        if (disabledReason) {
+          option.prepend = (
+            <>
```

Comment:
Does this mean the tooltip is only on prepend icon? Is there any way to keep the tooltip on the entire value? Could `renderOption` still be used?

Reply:
it is not, it happens on any place because of the custom css we place there. It works exactly as before. We cannot use renderOption if we want to take advantage of internal combobox truncation.

Reply:
How difficult is it to do the truncation in renderOptions? I think its important to keep the tooltip target over the entire text, otherwise, its really difficult to trigger.

Reply:
This is how it works, sorry I wasn't clear before. I agree it's important to keep the tooltip target over the entire text and it is kept this way in this PR. Right now it doesn't make sense to do the truncation in `renderOption` because the implementation is not straightforward and it's complex (the problem is not truncation but proper selecting text when searching through it). Please take a look at this video:

https://github.com/elastic/kibana/assets/4283304/607bb18d-12a9-44aa-8f41-22a0577120a9

Reply:
Thanks for clarifying